implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTCallerPrincipalFactory.java

```diff
@@ -16,11 +16,14 @@
  */
 package io.smallrye.jwt.auth.principal;
 
+import javax.enterprise.context.ApplicationScoped;
+
 import org.jose4j.jwt.consumer.JwtContext;
 
 /**
  * A default implementation of the abstract JWTCallerPrincipalFactory that uses the Keycloak token parsing classes.
  */
+@ApplicationScoped
```

Comment:
@andreas-eberle TCK is failing, perhaps there should be `JWTCallerPrincipalFactoryProducer` which will use the method you removed in `DefaultJWTParser` to produce a default factory, and the custom factories will be injected with the producers with `@Alternative` ?
Please also rebase
Thanks

Reply:
I was able to fix it by using the same mechanism used for DefaultJwtParser in `SmallRyeJWTAuthCDIExtension.java`.

Reply:
@andreas-eberle thanks